.github/workflows/build.yml

```diff
@@ -228,13 +224,17 @@ jobs:
     - name: "Create build directory"
       run: mkdir build
 
+    - name: "Enable fatal debug assertions for non-release builds"
+      if: github.event_name != 'push' || !startsWith(github.event.ref, 'refs/tags/')
```

Comment:
I don't think that this is a good idea. IMHO we need a release candidate that is build with exact the final build scripts. If that is tested, we can do the tag. Otherwise we tag and pray that everything works. If not we have to mess around with the tags.